hypertrace-core-graphql-span-schema/src/main/java/org/hypertrace/core/graphql/span/joiner/multiple/span/MultipleSpanJoiner.java

```diff
@@ -0,0 +1,29 @@
+package org.hypertrace.core.graphql.span.joiner.multiple.span;
+
+import com.google.common.collect.ArrayListMultimap;
+import com.google.common.collect.ListMultimap;
+import io.reactivex.rxjava3.core.Single;
+import java.util.Collection;
+import java.util.List;
+import org.hypertrace.core.graphql.span.schema.Span;
+
+public interface MultipleSpanJoiner {
+
+  /** A NOOP joiner */
+  MultipleSpanJoiner NO_OP_JOINER =
+      new MultipleSpanJoiner() {
+        @Override
+        public <T> Single<ListMultimap<T, Span>> joinMultipleSpans(
+            Collection<T> joinSources, MultipleSpanIdGetter<T> multipleSpanIdGetter) {
+          return Single.just(ArrayListMultimap.create());
+        }
+      };
+
+  <T> Single<ListMultimap<T, Span>> joinMultipleSpans(
```

Comment:
Sorry I think you misunderstood my earlier comment :( I was suggesting that you could do something like below with only a few extra lines of code compared to what you had
```java

public interface SpanJoiner {
  <T> Single<ListMultimap<T, Span>> joinSpans(Collection<T> joinSources, MultipleSpanIdGetter<T> multipleSpanIdGetter);

  <T> Single<Map<T, Span>> joinSpan(Collection<T> joinSources, SpanIdGetter<T> spanIdGetter);

  @FunctionalInterface
  interface MultipleSpanIdGetter<T> {
    Single<List<String>> getSpanIds(T source);
  }

  @FunctionalInterface
  interface SpanIdGetter<T> {
    Single<String> getSpanId(T source);
  }
```

Reply:
Or even split the interface but implement them in the same impl - that might be easier, since that way your builder can correctly disambiguate whether it should look for `SPAN_KEY` vs `SPANS_KEY`. But then constructs the same builder and returns it under the appropriate interface.